File: netlogo-headless/src/main/prim/etc/_fput.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.core.{ LogoList, Pure }
import org.nlogo.nvm.{ Context, Reporter }

class _fput extends Reporter with Pure {
  override def report(context: Context): LogoList =
    report_1(context, args(0).report(context), argEvalList(context, 1))
  def report_1(context: Context, obj: AnyRef, list: LogoList): LogoList =
    list.fput(obj)
}

File: netlogo-gui/src/main/prim/etc/_fput.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc;

import org.nlogo.api.LogoException;
import org.nlogo.core.LogoList;
import org.nlogo.core.Syntax;
import org.nlogo.nvm.Reporter;

public final strictfp class _fput
    extends Reporter
    implements org.nlogo.core.Pure {


  @Override
  public Object report(final org.nlogo.nvm.Context context) throws LogoException {
    return report_1(context, args[0].report(context), argEvalList(context, 1));
  }

  public LogoList report_1(final org.nlogo.nvm.Context context, Object obj, LogoList list) {
    return list.fput(obj);
  }
}
